uptane-standard.md

```diff
@@ -868,32 +873,37 @@ In order to perform partial verification, an ECU SHALL perform the following ste
+3. Refresh the top-level metadata for the Director repository, by following the steps in {#refresh_toplevel_metadata}, for the first repository in the list of repositories in the mapping metadata.
+4. If this ECU is a primary ECU, then iterate through the image metadata listed in the Director repository's top-level Targets metadata, and **store** all image metadata.  The primary ECU SHOULD also ensure that the image metadata from the director repository doesn't contain any ECU identifiers for ECUs not actually present in the vehicle.
+5. If instead this ECU is a secondary ECU, then iterate through the image metadata listed in the Director repository's top-level Targets metadata, and **store** the first image metadata listed that includes this secondary ECU's ECU identifier and hardware ID.
+6. For each image name for which the previous two steps **stored** image metadata, obtain verified image metadata for that image by iterating over each mapping in the repository mapping metadata, in order, and performing the following steps for each mapping:
+    1. If the image name does not match the image paths listed in the mapping, skip to the next mapping.
+    2. For each repository listed in this mapping, if top-level metadata has not been refreshed for that repository using {#refresh_toplevel_metadata} in this iteration of the {#full_verification} procedure, refresh top-level metadata for that repository using the steps in {#refresh_toplevel_metadata}.
+    3. For each repository listed in this mapping, perform the procedure in {{resolve_delegations}}, beginning at the top-level Targets metadata in that repository, to find metadata for the image if it exists, and **store** it if it does.
+    4. Find the first set of at least the mapping's threshold number of repositories, including the Director repository, for which all of the following is true for the stored image metadata for each repository:
+        1. The non-custom Targets metadata (i.e., length and hashes) of the unencrypted image are the same.
+        2. If listed, the "MUST match" custom Targets metadata (e.g., hardware identifier and release counter) are the same.
+        3. Any ECU identifiers, if listed, are the same.
+    5. If there is no such set (of at least the mapping's threshold number of repositories meeting the criteria in the prior step), then check the terminating property on this mapping.  If the terminating value for this mapping is true, then abort the search for image metadata for this image, discarding metadata **stored** for this image, as there is no verifiable update for it, and proceed to the next image for which there is **stored** image metadata, if any.  If the terminating value for this mapping is false, continue the search for verified image metadata for this image by continuing to the next mapping, discarding the image metadata for this image from the repositories other than the Director repository.
```

Comment:
In order to maintain the semantics of the mapping order in the face of man-in-the-middle attacks that interfere with the obtaining metadata for a prior mapping, it may be necessary to also abort the search if ANY of the repositories in a prior mapping provide metadata for an image, even if the mapping is not terminating.  If anyone has thoughts, please LMK, but this is probably fine.

Reply:
I think this step works as is.

Reply:
I'm confused why "discarding the image metadata for this image from the repositories other than the Director repository." is necessary. If there are 4 repositories, the threshold is 3, and an image's metadata is available in repos 1, 2, and 4, then this would cause that case to fail, but I think it should succeed.

Reply:
IIUC, no.  Those are the instructions for jumping to the next mapping, once a mapping's threshold has already failed to be met (`If there is no such set (of at least the mapping's threshold number of repositories meeting the criteria in the prior step`...).  So imagine:
- mapping 1, not terminating, all images: repos Director, R2, R3, and R4 must agree (threshold 3, must include Director)
- mapping 2 (details unimportant)

While processing mapping 1, you reach that instruction to discard image metadata once the 3/4 threshold for mapping 1 has already failed to be met (after you've checked Director, R2, R3, and R4).  At that point, you keep the image metadata (hashes, length, custom) for the image from the Director repo, but not from R2, R3, and R4.  You then move on to mapping 2, because mapping 1 was not terminating.  The image metadata for the Director is the same and still must match.

It looks like this may be too confusing.  I'm looking through Jon's changes.  Hopefully things will be simpler there.